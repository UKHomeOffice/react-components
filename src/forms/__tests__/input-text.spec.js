import React from 'react'
import { render } from 'enzyme'
import {Input} from 'govuk-react-components';


describe('The Input Text Component', () => {

  it('should render correctly when a value is passed in', () => {
    expect(
      render(<Input label="ip" name="ip" value="someText" />)
    ).toMatchSnapshot()
  });

  it('should render a hint when a hint is passed in', () => {
    expect(
      render(<Input label="ip" name="ip" value="someText" hint="Testing hint" />)
    ).toMatchSnapshot();
  });

  it('should render an error when an error is passed in', () => {
    expect(
      render(<Input label="ip" name="ip" value="someText" error="Testing error" />)
    ).toMatchSnapshot();
  });

  it('should render as a checked input correctly', () => {
    expect(
      render(<Input label="ip" name="ip" value="someText" onChange={() => { }} />)
    ).toMatchSnapshot();
  });
})
