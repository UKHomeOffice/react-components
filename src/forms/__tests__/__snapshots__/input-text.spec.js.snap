// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Input Text Component should render a hint when a hint is passed in 1`] = `
<div
  class="form-group"
>
  <label
    class="form-label"
    for="ip"
  >
    <span
      class="form-label-bold"
    >
      ip
    </span>
    <span
      class="form-hint"
    >
      Testing hint
    </span>
  </label>
  <input
    class="form-control"
    id="ip"
    name="ip"
    type="text"
    value="someText"
  />
</div>
`;

exports[`The Input Text Component should render an error when an error is passed in 1`] = `
<div
  class="form-group form-group-error"
>
  <label
    class="form-label"
    for="ip"
  >
    <span
      class="form-label-bold"
    >
      ip
    </span>
    <span
      class="error-message"
    >
      Testing error
    </span>
  </label>
  <input
    class="form-control form-control-error"
    id="ip"
    name="ip"
    type="text"
    value="someText"
  />
</div>
`;

exports[`The Input Text Component should render as a checked input correctly 1`] = `
<div
  class="form-group"
>
  <label
    class="form-label"
    for="ip"
  >
    <span
      class="form-label-bold"
    >
      ip
    </span>
  </label>
  <input
    class="form-control"
    id="ip"
    name="ip"
    type="text"
    value="someText"
  />
</div>
`;

exports[`The Input Text Component should render correctly when a value is passed in 1`] = `
<div
  class="form-group"
>
  <label
    class="form-label"
    for="ip"
  >
    <span
      class="form-label-bold"
    >
      ip
    </span>
  </label>
  <input
    class="form-control"
    id="ip"
    name="ip"
    type="text"
    value="someText"
  />
</div>
`;
